components/canvas_traits/canvas.rs

```diff
@@ -16,12 +16,13 @@ pub enum FillRule {
     Evenodd,
 }
 
-#[derive(Clone, Deserialize, MallocSizeOf, PartialEq, Serialize)]
+#[derive(Clone, Deserialize, Eq, Hash, MallocSizeOf, PartialEq, Serialize)]
```

Comment:
nit: derive `Copy`? It's just a number anyway. That allows its users more flexibility.

Reply:
Done.